### DISCUSSION TITLE
Favorite Markdown link style

### DISCUSSION THREAD

**Discussion by jcbhmr** [spenserblack/spenserblack#42] [General]
_Spin off of https://github.com/spenserblack/actions-wiki/pull/13#discussion_r1137024000_

Mine is `[term]` and then a big block at the bottom of the entire document defining all the `[term]: https://` links. What's yours?

#### Advantages
1. You don't have to think of a short ID for each link, you just select the text you want to link
2. You don't have to define a URL right away (also true for `[term][term-id]` links)
3. You can have GitHub Copilot auto-generate a link block at the bottom of your document for all the `[term]`s in your document
4. You avoid the `problem][with-long-tokens]` because that 👈 is interpreted as one single "word" and will wrap as such:

```md
I hope that you have enjoyed this! Our [HTML
Specification-v2.0][whatwg-html-v2.0] is not yet ready for the prime time, so
try to be patient with us as we make sure that it meets all the necessary
```

#### Disadvantages
1. Don't know exactly what a `[term]` will actually _resolve to_ without resolving it in an IDE or manually looking at the bottom link table

**Comment by spenserblack** [comment]
When I'm writing something where the Markdown code doesn't really matter at all (like a comment in a GitHub issue):

```markdown
[blah](https://example.com)
```

Otherwise, usually this:

```markdown
Please see the [examples][examples].

[examples]: https://example.com/docs/examples/
```

I try to keep the link IDs short so they look fine with the rendered text all on one line.
That way I can change the rendered text without needing to change the `[foo]: //bar`.
Also for reusability:

```markdown
## FAQ

### Do you have examples?

[Yes!!!!!][examples]
```

Though, if the link is short and only gets used once, I'll get lazy and do the first :slightly_smiling_face:

**Reply by jcbhmr** [reply]
```md
Please see the [examples][examples].

[examples]: https://example.com/docs/examples/
```

_sputtering_ but what about DRY??? 🤣

```md
Please see the [examples].

[examples]: https://example.com/docs/examples/
```

**Reply by spenserblack** [reply]
I follow WET
- Write
- Everything
- Twice
